File: 04_CSV_TO_SHAPEFILE.py
import arcpy, os
csv_path = r"C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS/POINTS.csv"

# Define the output shapefile path
output_folder = r"C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS"
output_shapefile = os.path.join(output_folder, "points_data.shp")

# Define the coordinate fields and spatial reference (e.g., WGS 84 with EPSG 4326)
x_field = "X"  # Replace with the actual column name for longitude
y_field = "Y"    # Replace with the actual column name for latitude
spatial_ref = arcpy.SpatialReference(32717)  # WGS 84

# Convert the CSV to a shapefile
arcpy.management.XYTableToPoint(csv_path, output_shapefile,x_field,y_field,
    coordinate_system=spatial_ref)
print(f"Shapefile created at: {output_shapefile}")

# Load the new shapefile into the ArcGIS Pro map
aprx = arcpy.mp.ArcGISProject("CURRENT")  # Use the currently open project
map_obj = aprx.activeMap  # Get the active map
aprx.save() # Save changes to the ArcGIS Pro project
print("Shapefile loaded and added to the map successfully.")

File: 04_CSV_TO_SHAPEFILE/04_CSV_TO_SHAPEFILE.R
# Load required libraries
if (!requireNamespace("sf", quietly = TRUE)) install.packages("sf")
if (!requireNamespace("ggplot2", quietly = TRUE)) install.packages("ggplot2")

library(sf)
library(ggplot2)

# Define the path to the CSV file
csv_path <- "C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS/POINTS.csv"

# Load the CSV file into a data frame
points_data <- read.csv(csv_path)

# Inspect the data
head(points_data)

# Define the coordinate reference system (EPSG 32717 - WGS 84 / UTM zone 17S)
crs_epsg <- 32717

# Convert the data frame to an sf object (point geometry)
points_sf <- st_as_sf(points_data, coords = c("X", "Y"), crs = crs_epsg)

# Define the path to save the shapefile
output_shapefile <- "C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS/points_data.shp"

# Save the sf object as a shapefile
st_write(points_sf, output_shapefile, delete_layer = TRUE)

print(paste("Shapefile created at:", output_shapefile))

# Plot the points using ggplot2
ggplot() +
  geom_sf(data = points_sf, color = "red", size = 5) +
  theme_minimal() +
  labs(title = "Points from CSV as Shapefile", x = "Easting", y = "Northing")
